src/features/pastelExpertConsole/rpc-services/api.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import axios from 'axios'
+
+import store from '../../../redux/store'
+import { APIMethods, METHODS } from './utils'
+
+const getRPCConfig = () => {
+  const state = store.getState()
+  const { pastelConf } = state
+
+  return pastelConf || {}
+}
+
+const getMessage = (statusCode: number, isECONNREFUSED: boolean) => {
+  if (isECONNREFUSED) {
+    return 'Pastel could not find a daemon running, please check the logs!'
+  }
+
+  switch (statusCode) {
+    case 401:
+      return 'Not authorized to access RPC, please check your rpcuser and rpcpassword'
+    default:
+      return 'Something went wrong'
+  }
+}
+
+const apiMethods: any = {}
+
+const api: APIMethods = METHODS.reduce(
+  (obj, method) => ({
+    ...obj,
+    [method]: (params = []) => {
```

Comment:
Using `reduce` and spread operator will create a copy of the `apiMethods` object as many times as there are elements in the `METHODS` array, which is inefficient. Instead, we should use good old `forEarch` or even `for`.